pkg/trafficpolicy/trafficpolicy.go

```diff
@@ -27,3 +28,31 @@ func (in *InboundTrafficPolicy) AddRule(route RouteWeightedClusters, allowedServ
 		})
 	}
 }
+
+// AddRoute adds a route to an OutboundTrafficPolicy given an HTTP route match and weighted cluster. If a Route with the given HTTP route match
+//	already exists, an error will be returned. If a Route with the given HTTP route match does not exist,
+//	a Route with the given HTTP route match and weighted clusters will be added to the Routes on the OutboundTrafficPolicy
+func (out *OutboundTrafficPolicy) AddRoute(httpRouteMatch HTTPRouteMatch, weightedClusters ...service.WeightedCluster) error {
+	routeExists := false
+	wc := set.NewSet()
+	for _, c := range weightedClusters {
+		wc.Add(c)
+	}
+
+	for _, existingRoute := range out.Routes {
```

Comment:
What scenario is this for? Would it be possible that the weightedclusters passed in would contain new wc's that need to be added? 

Can you add a comment with the scenario?